pkg/dns/aws/dns.go

```diff
@@ -101,13 +101,26 @@ func NewProvider(config Config, operatorReleaseVersion string) (*Provider, error
 	if len(region) == 0 {
 		return nil, fmt.Errorf("region is required")
 	}
+	awsconfig := &aws.Config{}
+	awsconfig.Region = &region
+
+	// TODO: This API will only return hostedzone resources (which are global)
+	// when the region is forced to us-east-1. We don't yet understand why.
+	tagRegion := "us-east-1"
+
+	//If the region is in aws china, cn-north-1 or cn-northwest-1, we should:
+	// 1. hard code route53 api endpoint to https://api.route53.cn, as route53 is not GA in AWS China, and aws sdk didn't have the endpoint.
+	// 2. use the aws china regions to setup tagging api correctly in stead of "use-east-1"
+	if (region == "cn-north-1") || (region == "cn-northwest-1") {
```

Comment:
```
select region {
case endpoints.CnNorth1RegionID, endpoints.CnNorthwest1RegionID:
....
}
```
try and use the const from https://docs.aws.amazon.com/sdk-for-go/api/aws/endpoints/#pkg-constants

Reply:
Are you talking using "switch"?